pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -1564,6 +1564,21 @@ type S3Bucket struct {
 
 	// UseSDKCreds tells the driver to figure out credentials based on sdk defaults.
 	UseSDKCreds bool `json:"useSDKCreds,omitempty" protobuf:"varint,8,opt,name=useSDKCreds"`
+
+	// CreateBucketIfNotPresent tells the driver to attempt to create the S3 bucket for output artifacts, if it doesn't exist
+	CreateBucketIfNotPresent CreateS3BucketOptions `json:"createBucketIfNotPresent,omitempty" protobuf:"bytes,9,opt,name=createBucketIfNotPresent"`
```

Comment:
this should be a pointer type, i.e. `CreateBucketIfNotPresent * CreateS3BucketOptions ` - `omitempty` has no effect otherwise

Reply:
done